### PR TITLE
chore: add e2e tests for codegen add, both with and without apiId

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Adding a regression test for basic happy-case functionality from https://github.com/aws-amplify/amplify-category-api/issues/38 this issue. Verified that using CLI v8.1.0 will fail the second test here.

#### Issue #, if available
https://github.com/aws-amplify/amplify-category-api/issues/38

#### Description of how you validated changes
Ran test, and verified test failed for 8.1.0

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/codegen/add.ts [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
-  return new Promise((resolve, reject) => {
-    const run = spawn(getCLIPath(), ['codegen', 'add'], { cwd, stripColors: true });
-    if (!(settings.ios || settings.android)) {
-      run.wait('Choose the code generation language target').sendCarriageReturn();
-    }
+type AddCodegenSettings = {
+  ios?: boolean;
+  android?: boolean;
+  apiId?: string;
+};
```
Comment (alharris-at) [inline packages/amplify-e2e-tests/src/codegen/add.ts:7]: Most of this file is just style changes, I added this type for the input, and added lines 15-18 to support adding codegen with an external api id.

File: packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/* eslint-disable spellcheck/spell-checker */
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts:1]: nit: eslint disable comments for new files should be avoided

We can add a missing word to our dictionary with `yarn addwords`, or if there's a string that doesn't makes sense to add, disable a specific line only.

File: packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    await updateApiSchema(projRoot, projName, 'simple_model.graphql');
+    await addCodegen(projRoot, {});
+  });
+
+  it('allows adding codegen to a project without an api using an api id', async () => {
+    // Set up project 1 with API
+    const proj1Name = createRandomName();
+    await initJSProjectWithProfile(projRoot, { name: proj1Name });
+    await addApiWithoutSchema(projRoot);
+    // await addApiWithoutSchema(projRoot, { transformerVersion: 1 });
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts:48]: nit: remove if not needed

**Review by jhockett — APPROVED** [review]
LGTM, just a couple nitpicks

File: packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts (hunk 2 above)
Comment (alharris-at) [reply packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts:1]: Ahh, my bad. I'd added that during dev so I wasn't messing w/ the file until I was done, forgot to remove it.

File: packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts (hunk 3 above)
Comment (alharris-at) [reply packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts:48]: Thanks for catching!

File: packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    await amplifyPush(projRoot);
+
+    // Get API Id
+    const { GraphQLAPIIdOutput } = getProjectMeta(projRoot).api[proj1Name].output;
+    expect(GraphQLAPIIdOutput).toBeDefined();
+
+    // Setup Project 2
+    const proj2Name = createRandomName();
+    await initJSProjectWithProfile(projRootExternalApi, { name: proj2Name });
+    await addCodegen(projRootExternalApi, { apiId: GraphQLAPIIdOutput });
```
Comment (AaronZyLee) [inline packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts:57]:
```suggestion
    await expect(addCodegen(projRootExternalApi, { apiId: GraphQLAPIIdOutput })).resolves.not.toThrow();
```

File: packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      deleteProjectDir(root);
+    });
+  });
+
+  it('allows adding codegen to a project with api', async () => {
+    const projName = createRandomName();
+    await initJSProjectWithProfile(projRoot, { name: projName });
+    await addApiWithoutSchema(projRoot);
+    await updateApiSchema(projRoot, projName, 'simple_model.graphql');
+    await addCodegen(projRoot, {});
```
Comment (AaronZyLee) [inline packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts:39]:
```suggestion
    await expect(addCodegen(projRoot, {})).resolves.not.toThrow();
```

**Review by AaronZyLee — APPROVED** [review]
Nit: add some suggestions for `expect`. LGTM 🏞️

File: packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts (hunk 4 above)
Comment (alharris-at) [reply packages/amplify-e2e-tests/src/__tests__/codegen/add_codegen.ts:57]: Great! I'll verify these work as expected, then push w/ that suggestion added.

**Pull request merged**